syntax/std_seq_array_helper.go

```diff
@@ -145,15 +144,35 @@ func arrayHasSuffix(suffix rel.Value, subject rel.Array) rel.Value {
 	return rel.NewBool(true)
 }
 
-func convert2Array(val rel.Value) rel.Array {
-	switch val := val.(type) {
-	case rel.Array:
-		return val
-	case rel.GenericSet:
-		valArray, _ := rel.AsArray(val)
-		return valArray
+func arrayTrimPrefix(prefix rel.Value, subject rel.Array) rel.Value {
+	if prefix.IsTrue() && subject.IsTrue() && arrayHasPrefix(prefix, subject).IsTrue() {
+		if prefixLen := prefix.(rel.Array).Count(); prefixLen <= subject.Count() {
```

Comment:
this check doesn't seem necessary, if `arrayHasPrefix` is true, then `prefixLen <= subject.Count()` is always true